library/src/test/java/com/novoda/downloadmanager/CallbackThrottleByProgressIncreaseTest.java

```diff
@@ -0,0 +1,67 @@
+package com.novoda.downloadmanager;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+public class CallbackThrottleByProgressIncreaseTest {
+
+    private static final int DOWNLOAD_PERCENTAGE = 75;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private final DownloadBatchCallback downloadBatchCallback = mock(DownloadBatchCallback.class);
+    private final DownloadBatchStatus downloadBatchStatus = mock(DownloadBatchStatus.class);
+
+    private CallbackThrottleByProgressIncrease callbackThrottleByProgressIncrease;
+
+    @Before
+    public void setUp() {
+        callbackThrottleByProgressIncrease = new CallbackThrottleByProgressIncrease();
+    }
+
+    @Test
+    public void throwsException_whenStoppingUpdates_withoutSettingCallback() {
+        thrown.expect(NullPointerException.class);
+
+        callbackThrottleByProgressIncrease.stopUpdates();
+    }
+
+    @Test
+    public void doesNothing_whenUpdating_withoutSettingCallback() {
+        callbackThrottleByProgressIncrease.update(downloadBatchStatus);
+
+        verifyZeroInteractions(downloadBatchCallback, downloadBatchStatus);
```

Comment:
hang on, does this check it doesn't call a callback you never set? If I get that right that doesn't sound like a test we should have. If you were just verifying it doesn't blow up, then that shouldn't need this bit

Reply:
This verifies that there are zero interactions with any mocks if a callback is not set

Reply:
but you gave it no reference to the callback, right? Like, if I instantiate an object and *don't* give it a collaborator, I don't need to check it doesn't call that collaborator. That would be magic if it did. Maybe I'm missing something

Reply:
yeah I think this test is not necessary, either:

- remove the test
- remove the assertion to verify that the CallbackThrottleByProgressIncrease doesn't throw an exception
- throw an exception if a callback is not set and we call update, then assert that

Reply:
I'm reluctant to change the code this early on considering I don't know enough about it. I'll remove this test for now and when more is known potentially retest this class.